docs/reference/api/rtos/EventQueue.md

```diff
@@ -1,11 +1,32 @@
 ## EventQueue
 
-[Add description here.]
+The EventQueue class provides a flexible queue for scheduling events. You can use the EventQueue class for synchronization between multiple threads, or to move events out of interrupt context (deferred execution of time consuming or non-ISR safe operations).
+
+The EventQueue class is thread and ISR safe.
+
+You can use the `dispatch` and `dispatch_forever` APIs to execute pending events. `break_dispatch` is to terminate the execution of events in the specified EventQueue. 
 
 ### EventQueue class reference
 
 [![View code](https://www.mbed.com/embed/?type=library)](https://os.mbed.com/docs/v5.6/mbed-os-api-doxy/classevents_1_1_event_queue.html)
 
-### EventQueue example
+### EventQueue example: deferring from interrupt context
+
+The code executes two handler functions (`rise_handler` and `fall_handler`) in two different contexts:
+
+1. In interrupt context when a rising edge is detected on `SW2` (`rise_handler`).
+2. In the context of the event loop's thread function when a falling edge is detected on `SW2` (`fall_handler`). `queue.event()` is called with `fall_handler` as an argument to specify that `fall_handler` runs in user context instead of interrupt context.
```

Comment:
Query: Would it be correct to say "`fall_handler` calls `queue.event()` as an argument to specify..."?

Reply:
No, fall_handler function is an argument to queue.event().
We can rephrase it as, 
You can use 'fall_handler` function as argument to `queue.event()` if you want to run it in user context instead of interrupt context.